tcl.html.markdown

```diff
@@ -30,193 +30,253 @@ Various features of Tcl stand out:
-set greeting $greeting1$greeting2[set greeting3]
+# Left and right brackets embed a script to be evaluated for a result to
+# substitute into the word:
+set greeting $part1$part2[set part3]
 
 
-# Command substitution should really be called script substitution, because an
-# entire script, not just a command, can be placed between the brackets. The
-# "incr" command increments the value of a variable and returns its value.
-
-set i 0
+# An embedded script may be composed of multiple commands, the last of which provides
+# the result for the substtution:
 set greeting $greeting[
-	incr i
-	incr i
-	incr i
+    incr i
+    incr i
+    incr i
 ]
-# i is now 3
+puts $greeting ;# The output is "Salutations3" 
 
-# backslash suppresses the special meaning of characters
+# Every word in a command is a string, including the name of the routine, so
+# substitutions can be used on it as well. Given this variable
+# assignment,
+set action pu
+
+# , the following three commands are equivalent:
+puts $greeting
+${action}ts $greeting 
+[set action]ts $greeting
+
+
+# backslash suppresses the special meaning of characters:
 set amount \$16.42
 
 
-# backslash adds special meaning to certain characters
+# backslash adds special meaning to certain characters:
 puts lots\nof\n\n\n\n\n\nnewlines
 
 
 # A word enclosed in braces is not subject to any special interpretation or
 # substitutions, except that a backslash before a brace is not counted when
-# looking for the closing brace
+# looking for the closing brace:
 set somevar {
     This is a literal $ sign, and this \} escaped
     brace remains uninterpreted
 }
 
 
 # In a word enclosed in double quotes, whitespace characters lose their special
-# meaning
+# meaning:
 set name Neo
 set greeting "Hello, $name"
 
 
-#variable names can be any string
+# A variable name can be any string:
 set {first name} New
 
 
-# The brace form of variable substitution handles more complex variable names
+# The braced form of variable substitution handles more complex variable names:
 set greeting "Hello, ${first name}"
 
 
-# The "set" command can always be used instead of variable substitution
+# "set" can always be used instead of variable substitution, and can handle all
+# variable names:
 set greeting "Hello, [set {first name}]"
 
 
-# To promote the words within a word to individual words of the current
-# command, use the expansion operator, "{*}".
-```
-```tcl
-set {*}{name Neo}
-
-# is equivalent to
+# To unpack a list into the command, use the expansion operator, "{*}".  These
+# two commands are equivalent:
 set name Neo
+set {*}{name Neo}
 
 
 # An array is a special variable that is a container for other variables.
 set person(name) Neo
-set person(gender) male
+set person(destiny) {The One}
 set greeting "Hello, $person(name)"
 
 
-# A namespace holds commands and variables
+# "variable" can be used to declare or set variables. In contrast with "set",
```

Comment:
I actually tried something like that at first, but found the propagation of "the command 'commandname'" throughout the document to be more verbose and distracting.  "variable" is an edge case where the name of the command happens to be the name of an overloaded word.  My vote is to leave it alone.  It may give the reader a slight pause, but the ensuing example, along with the consistent treatment of command names in other contents, clear it up pretty quickly.